core/domain/email_services.py

```diff
@@ -69,3 +75,165 @@ def get_feedback_thread_reply_info_by_user_and_thread_ids(user_id, thread_id):
+    """
+    if not isinstance(email_address, python_utils.BASESTRING):
+        return False
+
+    stripped_address = email_address.strip()
+    if not stripped_address:
+        return False
+    # Regex for a valid email.
+    regex = r'^.+@(\[?)[a-zA-Z0-9-.]+.([a-zA-Z]{2,3}|[0-9]{1,3})(]?)$'
+    return re.search(regex, email_address)
+
+
+def is_sender_email_valid(sender_email):
+    """Gets the sender_email address and validates that it is of the form
+    'SENDER_NAME <SENDER_EMAIL_ADDRESS>' or 'email_address'.
+
+    Args:
+        sender_email: str. The email address of the sender.
+
+    Returns:
+        bool. Whether the sender_email is valid.
+    """
+    # Checks the case where sender_email is of the form 'Jane <EMAIL>'.
+    sender_email_with_only_first_name_regex = (
+        r'^[a-zA-Z._]+ <.+@(\[?)[a-zA-Z0-9-.]+.([a-zA-Z]{2,3}|[0-9]{1,3})' +
+        '(]?)>$')
+    # Checks the case where sender_email is of the form 'Jane Doe <EMAIL>'.
+    sender_email_with_full_name_regex = (
+        r'^[a-zA-Z._]+ [a-zA-Z._]+ <.+@(\[?)[a-zA-Z0-9-.]+.([a-zA-Z]{2,3}' +
+        r'|[0-9]{1,3})(]?)>$')
+    return is_email_valid(sender_email) or (
+        re.search(sender_email_with_only_first_name_regex, sender_email)) or (
+            re.search(sender_email_with_full_name_regex, sender_email))
+
+
+def send_mail(
+        sender_email, recipient_email, subject, plaintext_body,
+        html_body, bcc_admin=False, reply_to_id=None):
+    """Sends an email.
+
+    In general this function should only be called from
+    email_manager._send_email().
+
+    Args:
+        sender_email: str. The email address of the sender. This should be in
+            the form 'SENDER_NAME <SENDER_EMAIL_ADDRESS>' or
+            'SENDER_EMAIL_ADDRESS'. Format must be utf-8.
+        recipient_email: str. The email address of the recipient. Format must
+            be utf-8.
+        subject: str. The subject line of the email. Format must be utf-8.
+        plaintext_body: str. The plaintext body of the email. Format must be
+            utf-8.
+        html_body: str. The HTML body of the email. Must fit in a datastore
+            entity. Format must be utf-8.
+        bcc_admin: bool. Whether to bcc feconf.ADMIN_EMAIL_ADDRESS on the email.
+        reply_to_id: str|None. The unique id of the sender. Format must be
+            utf-8.
+
+    Raises:
+        Exception: The configuration in feconf.py forbids emails from being
+            sent.
+        Exception: If any recipient email addresses are malformed.
+        Exception: If any sender email addresses are malformed.
+        Exception: If the email was not sent correctly. In other words, the
+            send_email_to_recipients() function returned False
+            (signifying API returned bad status code).
+    """
+    if not feconf.CAN_SEND_EMAILS:
+        raise Exception('This app cannot send emails to users.')
+
+    if not is_email_valid(recipient_email):
```

Comment:
One note -- if you don't expect these helper functions to be called from outside this module, prefix them with _.

(If the only thing calling them is tests, you can still test them by testing this function with appropriate arguments.)